### PR TITLE
Add public comments for bundles

### PR DESCRIPTION
**Pull request by wu-hui**

### CODE REVIEW

File: Firestore/Source/Public/FirebaseFirestore/FIRFirestore.h [hunk 1]
```diff
@@ -261,23 +261,67 @@ NS_SWIFT_NAME(Firestore)
 
 #pragma mark - Bundles
 
+/**
+ * Loads a Firestore bundle into the local cache.
+ *
+ * @param bundleData Data from the bundle to be loaded.
+ * @return A `FIRLoadBundleTask` (`LoadBundleTask` in Swift), which allow registering observers
```
Comment (var-const) [inline Firestore/Source/Public/FirebaseFirestore/FIRFirestore.h:268]: Nits:
* s/allow/allows/;
* I think the comma before `which` is unnecessary;
* s/registering/registered/;
* I think the comma after `updates` is unnecessary.

File: Firestore/Source/Public/FirebaseFirestore/FIRFirestore.h [hunk 2]
```diff
@@ -261,23 +261,67 @@ NS_SWIFT_NAME(Firestore)
...
+ * to receive progress updates, and completion or error events.
+ */
 - (FIRLoadBundleTask *)loadBundle:(NSData *)bundleData NS_SWIFT_NAME(loadBundle(_:));
 
+/**
+ * Loads a Firestore bundle into the local cache.
+ *
+ * @param bundleData Data from the bundle to be loaded.
+ * @param completion A block to execute when the loading is in a final state, the `error` parameter
```
Comment (var-const) [inline Firestore/Source/Public/FirebaseFirestore/FIRFirestore.h:277]: I _think_ it should be `s/the loading/loading/`. However, it would be great to check this with the tech writer (I'm not a native speaker and can easily be mistaken).

Comment (var-const) [inline Firestore/Source/Public/FirebaseFirestore/FIRFirestore.h:277]: I think this would flow better if you split this in two sentences:
```
A block to execute when the loading is in a final state. The `error` parameter of the block...
```

File: Firestore/Source/Public/FirebaseFirestore/FIRFirestore.h [hunk 3]
```diff
@@ -261,23 +261,67 @@ NS_SWIFT_NAME(Firestore)
...
+ * of the block will be set if it is due to an error. If observers are registered to the returning
```
Comment (var-const) [inline Firestore/Source/Public/FirebaseFirestore/FIRFirestore.h:278]: Nit: `if it is` seems a little vague -- perhaps something like `if the block is invoked due to an error`?

File: Firestore/Source/Public/FirebaseFirestore/FIRFirestore.h [hunk 4]
```diff
@@ -261,23 +261,67 @@ NS_SWIFT_NAME(Firestore)
...
+ * `FIRLoadBundleTask`, this block will be called after all observers are notified.
```
Comment (var-const) [inline Firestore/Source/Public/FirebaseFirestore/FIRFirestore.h:279]: I _think_ it would read slightly better as `to the FIRLoadBundleTask being returned`, but it's also best to check with the tech writer.

File: Firestore/Source/Public/FirebaseFirestore/FIRFirestore.h [hunk 5]
```diff
@@ -261,23 +261,67 @@ NS_SWIFT_NAME(Firestore)
...
+ */
 - (FIRLoadBundleTask *)loadBundle:(NSData *)bundleData
                        completion:(nullable void (^)(FIRLoadBundleTaskProgress *_Nullable progress,
                                                      NSError *_Nullable error))completion
     NS_SWIFT_NAME(loadBundle(_:completion:));
 
+/**
+ * Loads a Firestore bundle into the local cache.
+ *
+ * @param bundleStream A input stream from which the bundle can be read.
```
Comment (var-const) [inline Firestore/Source/Public/FirebaseFirestore/FIRFirestore.h:291]: Nit: `s/A input/An input/` (here and below).

File: Firestore/Source/Public/FirebaseFirestore/FIRFirestore.h [hunk 6]
```diff
@@ -261,23 +261,67 @@ NS_SWIFT_NAME(Firestore)
...
+ * to receive progress updates, and completion or error events.
+ */
 - (FIRLoadBundleTask *)loadBundleStream:(NSInputStream *)bundleStream
                              completion:
                                  (nullable void (^)(FIRLoadBundleTaskProgress *_Nullable progress,
                                                     NSError *_Nullable error))completion
     NS_SWIFT_NAME(loadBundle(_:completion:));
 
+/**
+ * Reads a `FIRQuery` (`Query` in Swift) from local cache, identified by the given name.
```
Comment (var-const) [inline Firestore/Source/Public/FirebaseFirestore/FIRFirestore.h:314]: Nit: I _think_ `local cache` needs an article -- i.e. `from the local cache` (applies throughout this comment).

File: Firestore/Source/Public/FirebaseFirestore/FIRFirestore.h [hunk 7]
```diff
@@ -261,23 +261,67 @@ NS_SWIFT_NAME(Firestore)
...
+ *
+ * The named queries are packaged into bundles on the server side (along with resulting documents)
+ * and loaded into local cache using `loadBundle`. Once in local cache, you can use this method to
+ * extract a query by name.
+ *
+ * @param completion A block to execute with the query read from local cache. If no query can be
+ * found, it's parameter will be `nil`.
```
Comment (var-const) [inline Firestore/Source/Public/FirebaseFirestore/FIRFirestore.h:321]: Nit: `s/it's/its/` (`it's` is a contraction of `it is`; `its` is the possessive form of `it`). However, I _think_ it might be better to replace `its` with `the`.

File: Firestore/Source/Public/FirebaseFirestore/FIRFirestore.h [hunk 8]
```diff
@@ -261,23 +261,67 @@ NS_SWIFT_NAME(Firestore)
...
 - (FIRLoadBundleTask *)loadBundleStream:(NSInputStream *)bundleStream
                              completion:
                                  (nullable void (^)(FIRLoadBundleTaskProgress *_Nullable progress,
                                                     NSError *_Nullable error))completion
     NS_SWIFT_NAME(loadBundle(_:completion:));
 
+/**
+ * Reads a `FIRQuery` (`Query` in Swift) from local cache, identified by the given name.
+ *
+ * The named queries are packaged into bundles on the server side (along with resulting documents)
```
Comment (var-const) [inline Firestore/Source/Public/FirebaseFirestore/FIRFirestore.h:316]: Nits: I _think_ `named queries` doesn't need an article. I suspect `resulting documents` needs `the`, but it's best to check.

File: Firestore/Source/Public/FirebaseFirestore/FIRLoadBundleTask.h [hunk 9]
```diff
@@ -18,6 +18,12 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+/**
+ * Represents the state of bundle loading tasks.
+ *
+ * Both `FIRLoadBundleTaskStateError` and `FIRLoadBundleTaskStateSuccess` are final states: task
+ * will abort or complete and there will be no more updates after they are reported.
```
Comment (var-const) [inline Firestore/Source/Public/FirebaseFirestore/FIRLoadBundleTask.h:25]: Nit: `will abort or complete` seems to indicate this actions will happen after these states are reported, but IIUC, the task will already be in one of these states. Consider something like `a task will be in either the aborted or the completed state and...`.

I also think `task` needs an article, but I'm not 100% sure if it's `a` or `the` (leaning towards `a`).

File: Firestore/Source/Public/FirebaseFirestore/FIRLoadBundleTask.h [hunk 10]
```diff
@@ -28,28 +34,57 @@ typedef NS_ENUM(NSInteger, FIRLoadBundleTaskState) {
...
+
+/** The total number of bytes in the bundle. 0 if the bundle failed to parse. */
 @property(readonly, nonatomic) NSInteger totalBytes;
 
+/** The current state of `FIRLoadBundleTask` (`LoadBundleTask` in Swift). */
 @property(readonly, nonatomic) FIRLoadBundleTaskState state;
 
 @end
 
+/** A handle associated with registered observers, that can be used to remove them. */
```
Comment (var-const) [inline Firestore/Source/Public/FirebaseFirestore/FIRLoadBundleTask.h:58]: Nit: I think the comma is unnecessary.

File: Firestore/Source/Public/FirebaseFirestore/FIRLoadBundleTask.h [hunk 11]
```diff
@@ -28,28 +34,57 @@ typedef NS_ENUM(NSInteger, FIRLoadBundleTaskState) {
...
 typedef NSInteger FIRLoadBundleHandle NS_SWIFT_NAME(LoadBundleHandle);
 
+/**
+ * Represents the task of loading a Firestore bundle. Observers can be registered with this task to
+ * observe progresses of bundle loading, as well as task completion and error events.
```
Comment (var-const) [inline Firestore/Source/Public/FirebaseFirestore/FIRLoadBundleTask.h:63]: Hmm, I'd expect `progresses` to be singular -- `observe the progress` (perhaps `observe the bundle loading progress`), but better check (I think both are correct, it's just a question of which flows better).

File: Firestore/Source/Public/FirebaseFirestore/FIRLoadBundleTask.h [hunk 12]
```diff
@@ -28,28 +34,57 @@ typedef NS_ENUM(NSInteger, FIRLoadBundleTaskState) {
...
+ */
 NS_SWIFT_NAME(LoadBundleTask)
 @interface FIRLoadBundleTask : NSObject
 
-- (FIRLoadBundleHandle)observeWithHandler:(void (^)(FIRLoadBundleTaskProgress *progress))handler;
+/**
+ * Registers an observer to observe the progress updates, completion or error events.
+ *
+ * @return A handle to the registered observer, which can be used to remove the observer once it is
```
Comment (var-const) [inline Firestore/Source/Public/FirebaseFirestore/FIRLoadBundleTask.h:71]: I _think_ the comma is not necessary.

File: Firestore/Source/Public/FirebaseFirestore/FIRLoadBundleTask.h [hunk 13]
```diff
@@ -28,28 +34,57 @@ typedef NS_ENUM(NSInteger, FIRLoadBundleTaskState) {
...
+ * no longer needed.
+ */
+- (FIRLoadBundleHandle)observeWith:(void (^)(FIRLoadBundleTaskProgress *progress))handler
```
Comment (var-const) [inline Firestore/Source/Public/FirebaseFirestore/FIRLoadBundleTask.h:74]: Hmm, I'm a little surprised `Handler` is removed from the method name. I'm not very familiar with the Objective-C naming conventions, but I thought the parameter name is generally repeated in the function name. What is the reason for this change?

**Review by var-const — COMMENTED** [review]
I think you previously mentioned working with a tech writer on these comments -- can you please involve the tech writer in the review? I left some comments, but I'm not a native speaker, so it's best to double-check them with the tech writer if possible (especially the comments about articles).

File: Firestore/Source/Public/FirebaseFirestore/FIRLoadBundleTask.h (hunk 12 above)
Comment (wu-hui) [reply Firestore/Source/Public/FirebaseFirestore/FIRLoadBundleTask.h]: Done.

File: Firestore/Source/Public/FirebaseFirestore/FIRLoadBundleTask.h (hunk 13 above)
Comment (wu-hui) [reply Firestore/Source/Public/FirebaseFirestore/FIRLoadBundleTask.h:75]: It's mainly because I found it very awkward to have both `handler` in code and `observer` in comment referring to pretty much the same thing.

I am open to suggestions.

Comment (wu-hui) [reply Firestore/Source/Public/FirebaseFirestore/FIRLoadBundleTask.h:75]: What do you think about `addObserver` instead? Also I think `FIRLoadBundleHandle` should be `FIRLoadBundleObserverHandle`.

File: Firestore/Source/Public/FirebaseFirestore/FIRFirestore.h [hunk 14]
```diff
@@ -265,20 +265,20 @@ NS_SWIFT_NAME(Firestore)
...
  * Loads a Firestore bundle into the local cache.
  *
  * @param bundleData Data from the bundle to be loaded.
- * @param completion A block to execute when the loading is in a final state, the `error` parameter
- * of the block will be set if it is due to an error. If observers are registered to the returning
- * `FIRLoadBundleTask`, this block will be called after all observers are notified.
- * @return A `FIRLoadBundleTask` (`LoadBundleTask` in Swift), which allow registering observers
- * to receive progress updates, and completion or error events.
+ * @param completion A block to execute when loading is in a final state. The `error` parameter
+ * of will be set if the block is invoked due to an error. If observers are registered to the
```
Comment (var-const) [inline Firestore/Source/Public/FirebaseFirestore/FIRFirestore.h:278]: Nit: `s/of will/will/`

File: Firestore/Source/Public/FirebaseFirestore/FIRLoadBundleTask.h (hunk 13 above)
Comment (var-const) [reply Firestore/Source/Public/FirebaseFirestore/FIRLoadBundleTask.h:75]: Agree on both (you probably need to talk to the API council, but IIUC, you're already on that)

**Review by var-const — APPROVED** [review]

File: Firestore/Source/Public/FirebaseFirestore/FIRFirestore.h (hunk 14 above)
Comment (wu-hui) [reply Firestore/Source/Public/FirebaseFirestore/FIRFirestore.h]: Done.

File: Firestore/Source/Public/FirebaseFirestore/FIRFirestore.h [hunk 15]
```diff
@@ -261,23 +261,67 @@ NS_SWIFT_NAME(Firestore)
...
+ * @param bundleStream An input stream from which the bundle can be read.
+ * @return A `FIRLoadBundleTask` (`LoadBundleTask` in Swift) which allows registered observers
+ * to receive progress updates and completion or error events.
+ */
 - (FIRLoadBundleTask *)loadBundleStream:(NSInputStream *)bundleStream NS_SWIFT_NAME(loadBundle(_:));
 
+/**
+ * Loads a Firestore bundle into the local cache.
+ *
+ * @param bundleStream A input stream from which the bundle can be read.
```
Comment (markarndt) [inline Firestore/Source/Public/FirebaseFirestore/FIRFirestore.h:300]: "An input stream...."

File: Firestore/Source/Public/FirebaseFirestore/FIRFirestore.h [hunk 16]
```diff
@@ -261,23 +261,67 @@ NS_SWIFT_NAME(Firestore)
...
+ * @param completion A block to execute when the loading is in a final state, the `error` parameter
```
Comment (markarndt) [inline Firestore/Source/Public/FirebaseFirestore/FIRFirestore.h:301]: "...in a final state. The error block..."
(not a comma)

File: Firestore/Source/Public/FirebaseFirestore/FIRFirestore.h [hunk 17]
```diff
@@ -261,23 +261,67 @@ NS_SWIFT_NAME(Firestore)
...
+ */
 - (FIRLoadBundleTask *)loadBundle:(NSData *)bundleData NS_SWIFT_NAME(loadBundle(_:));
 
+/**
+ * Loads a Firestore bundle into the local cache.
+ *
+ * @param bundleData Data from the bundle to be loaded.
+ * @param completion A block to execute when loading is in a final state. The `error` parameter
+ * will be set if the block is invoked due to an error. If observers are registered to the
+ * `FIRLoadBundleTask` being returned, this block will be called after all observers are notified.
```
Comment (markarndt) [inline Firestore/Source/Public/FirebaseFirestore/FIRFirestore.h:279]: Delete "...being returned,..."

**Review by markarndt — COMMENTED** [review]
Comments inline...

File: Firestore/Source/Public/FirebaseFirestore/FIRFirestore.h (hunk 17 above)
Comment (wu-hui) [reply Firestore/Source/Public/FirebaseFirestore/FIRFirestore.h]: Done.

File: Firestore/Source/Public/FirebaseFirestore/FIRFirestore.h (hunk 15 above)
Comment (wu-hui) [reply Firestore/Source/Public/FirebaseFirestore/FIRFirestore.h]: Done.

File: Firestore/Source/Public/FirebaseFirestore/FIRFirestore.h (hunk 16 above)
Comment (wu-hui) [reply Firestore/Source/Public/FirebaseFirestore/FIRFirestore.h]: Done.

**Review by markarndt — APPROVED** [review]
